samples/go/csv/read.go

```diff
@@ -157,18 +157,34 @@ func readFieldsFromRow(row []string, headers []string, fieldOrder []int, kindMap
 	return fields
 }
 
+func primaryKeyFromFields(fields types.ValueSlice, fieldOrder, pkIndices []int) (types.ValueSlice, types.Value) {
+	numPrimaryKeys := len(pkIndices)
+	d.Chk.True(numPrimaryKeys >= 1, "No primary key defined when reading into map")
+
+	if numPrimaryKeys == 1 {
```

Comment:
Since this is so common, it might be best to make this primaryKeyFromFields call an internal func variable inside ReadToMap - if numPrimaryKeys == 1 then the func is simply line 165. Otherwise the func is lines 168-178. Maybe, not sure but it seems pretty clean and then you also do not have to pass all of the arguments again since it could be a closure. No big deal if you disagree here, just a suggestion.